types/src/v0/impls/instance_state.rs

```diff
@@ -325,6 +347,43 @@ impl Upgrade {
             },
         }
     }
+    pub fn pos_view_based(address: Address) -> Upgrade {
+        let chain_config = ChainConfig {
+            base_fee: 0.into(),
+            stake_table_contract: Some(address),
```

Comment:
More descriptive name? For example:
```suggestion
    pub fn pos_view_based(stake_table_contract: Address) -> Upgrade {
        let chain_config = ChainConfig {
            base_fee: 0.into(),
            stake_table_contract: Some(stake_table_contract),
```